### PR TITLE
OHRM5X-341: [PIM - Qualifications] [License] Unable to select future years in expiry date field.

### PR DESCRIPTION
**Pull request by ireshe**
Follwoing issues are fixed from the this PR
 OHRM5X-341 [PIM - Qualifications] [License] Unable to select future years in expiry date field
OHRM5X-333 Phone number field accepts letters and special characters if the trailing character is an acceptable character

### CODE REVIEW

File: symfony/client/src/core/util/helper/future-years.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+export const futureYears = function() {
+  const numberOfYears = new Array(120);
```
Comment (RajithaKumara) [inline symfony/client/src/core/util/helper/future-years.ts:20]: Oops. it's static range in 1971 to 2091?

File: symfony/client/src/core/util/validation/__tests__/rules.spec.ts [hunk 2]
```diff
@@ -154,3 +159,60 @@ describe('core/util/validation/rules::endDateShouldBeAfterStartDate', () => {
     expect(result).toBe('End date should be after start date');
   });
 });
+
+describe('core/util/validation/rules::validPhoneNumberFormat', () => {
```
Comment (RajithaKumara) [inline symfony/client/src/core/util/validation/__tests__/rules.spec.ts:163]: @samanthajayasinghe any feedback on phone number validation for FE.
Currently BE don't have strict validation

File: symfony/client/src/orangehrmPimPlugin/components/EditLicense.vue [hunk 3]
```diff
@@ -88,6 +89,7 @@ import {
   shouldNotExceedCharLength,
   endDateShouldBeAfterStartDate,
 } from '@orangehrm/core/util/validation/rules';
+import {futureYears} from '@orangehrm/core/util/helper/future-years';
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmPimPlugin/components/EditLicense.vue:92]: `futureYears` not actually futures years here. Seems here we have year range

File: symfony/client/src/orangehrmPimPlugin/components/SaveLicense.vue [hunk 4]
```diff
@@ -61,6 +61,7 @@
               v-model="license.expiryDate"
               :rules="rules.expiryDate"
               type="date"
+              :years="endYear"
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmPimPlugin/components/SaveLicense.vue:64]: `endYear` also not make sense for year array

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/client/src/core/util/validation/__tests__/rules.spec.ts (hunk 2 above)
Comment (samanthajayasinghe) [reply symfony/client/src/core/util/validation/__tests__/rules.spec.ts:163]: don't validate the phone number .. it's change county to country

Comment (ireshe) [reply symfony/client/src/core/util/validation/__tests__/rules.spec.ts:163]: hi @samanthajayasinghe we only validatate  numbers and  + - / ( ) and check if it's less than 25, so it won't be problem right ? although there are different country codes and styles

File: symfony/client/src/orangehrmPimPlugin/components/SaveLicense.vue (hunk 4 above)
Comment (ireshe) [reply symfony/client/src/orangehrmPimPlugin/components/SaveLicense.vue:64]: use yearArray instead

File: symfony/client/src/core/util/helper/future-years.ts (hunk 1 above)
Comment (ireshe) [reply symfony/client/src/core/util/helper/future-years.ts:20]: change the function to get the current year and display both 50 years from past to present.

File: symfony/client/src/core/util/helper/year-range.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+export const YearRange = function() {
+  const numberOfYears = new Array(100);
+  const currentTime = new Date();
+  for (let i = 0; i < 100; i++) {
+    numberOfYears[i] = currentTime.getFullYear() - 50 + i;
```
Comment (Super-Chama) [inline symfony/client/src/core/util/helper/year-range.ts:23]: `fromYear - Math.floor(numberOfYears/2) + i` would be better.

File: symfony/client/src/core/util/helper/year-range.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+export const YearRange = function() {
+  const numberOfYears = new Array(100);
+  const currentTime = new Date();
+  for (let i = 0; i < 100; i++) {
```
Comment (Super-Chama) [inline symfony/client/src/core/util/helper/year-range.ts:22]: `for (let i = 0; i < numberOfYears.length; i++)`

**Review by Super-Chama — COMMENTED** [review]
These are some suggestions to improve the year range api.

File: symfony/client/src/core/util/helper/year-range.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+export const YearRange = function() {
```
Comment (Super-Chama) [inline symfony/client/src/core/util/helper/year-range.ts:19]: can we improve this function to accept some parameters, incase we have to use this function differently in future.

`function(range = 100, fromYear = new Date().getFullYear())`

Comment (RajithaKumara) [reply symfony/client/src/core/util/helper/year-range.ts:19]: +1
And make functiona nale camelCase

File: symfony/client/src/core/util/helper/year-range.ts (hunk 5 above)
Comment (ireshe) [reply symfony/client/src/core/util/helper/year-range.ts]: done

File: symfony/client/src/core/util/helper/year-range.ts (hunk 7 above)
Comment (ireshe) [reply symfony/client/src/core/util/helper/year-range.ts]: make it camelCase

**Pull request merged**